pkg/worker/log/watcher_linux.go

```diff
@@ -0,0 +1,110 @@
+package log
+
+import (
+	"context"
+	"log"
+	"syscall"
+	"unsafe"
+
+	"golang.org/x/sys/unix"
+)
+
+type fileLinuxEvent struct {
+	mask uint32
+}
+
+// Modified returns true if the event is syscall.IN_MODIFY.
+func (e *fileLinuxEvent) Modified() bool {
+	return e.mask&syscall.IN_MODIFY == syscall.IN_MODIFY
+}
+
+// Deleted returns true if the event is syscall.IN_DELETE
+func (e *fileLinuxEvent) Deleted() bool {
+	return e.mask&syscall.IN_DELETE == syscall.IN_DELETE
+}
+
+// wrapLinuxEvent create a new File event given an event mask.
+func wrapLinuxEvent(mask uint32) FileEvent {
+	return &fileLinuxEvent{
+		mask: mask,
+	}
+}
+
+// linuxWatcher Watcher implementation for Linux
+type linuxWatcher struct{}
+
+// NewWatcher instance
+func NewWatcher() Watcher {
+	return &linuxWatcher{}
+}
+
+// Watcher watch file system events and send them throught a channel.
+// See https://man7.org/linux/man-pages/man7/inotify.7.html
+func (e *linuxWatcher) Watch(ctx context.Context, path string, mask uint32) (eventchan chan FileEvent, err error) {
+	filed, err := syscall.InotifyInit()
+	if filed == -1 || err != nil {
```

Comment:
`syscall.InotifyInit` should return a non-nil error based on `errno`, so no need to check `filed` separately